src/js/jsx/search/SearchBar.jsx

```diff
@@ -250,41 +303,232 @@ define(function (require, exports, module) {
 
             var recentDocLabel = {
                 id: "recent-doc_header",
-                title: "Recent Documents",
+                title: HEADERS.RECENT_DOCS,
                 type: "header"
             };
 
-            recentDocMap.unshift(recentDocLabel);
+            return recentDocMap.unshift(recentDocLabel);
+        },
+
+        /**
+         * Make list of search category dropdown options based on header
+         * 
+         * @param {string} header, either "layer" or "document"
+         * @return {Array.<object>}
+         */
+        _getFilterOptions: function (header) {
+            if (header !== "layer" && header !== "document") {
+                return;
+            }
+            var categoryList = header === "document" ? ["current", "recent"] : layerLib.layerKinds,
+                categories = Immutable.fromJS(categoryList).filterNot(function (kind) {
```

Comment:
For clarity (and to avoid accidentally processing deep structures) we avoid using `Immutable.fromJS` and `toJS` in production code in favor of the more explicit constructor functions like `Immutable.List(categoryList)`.